url75694980d5_fabioalmeid_tp01ftc_tgz-pJ8-agentes_jade_Centralizador_ExemploParserGramaticaCentralizadorJ8/src/Centralizador/PrettyPrinter.java

```diff
@@ -0,0 +1,568 @@
+  {
+    pp(foo, 0);
+    trim();
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  public static String show(Centralizador.Absyn.Remedio foo)
+  {
+    sh(foo);
+    String temp = buf_.toString();
+    buf_.delete(0,buf_.length());
+    return temp;
+  }
+  /***   You shouldn't need to change anything beyond this point.   ***/
+
+  private static void pp(Centralizador.Absyn.Tarefa foo, int _i_)
+  {
+    if (foo instanceof Centralizador.Absyn.ETask)
+    {
+       Centralizador.Absyn.ETask _etask = (Centralizador.Absyn.ETask) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       pp(_etask.acao_, 0);
+       if (_i_ > 0) render(_R_PAREN);
+    }
+  }
+
+  private static void pp(Centralizador.Absyn.Acao foo, int _i_)
+  {
+    if (foo instanceof Centralizador.Absyn.EAction1)
+    {
+       Centralizador.Absyn.EAction1 _eaction1 = (Centralizador.Absyn.EAction1) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       pp(_eaction1.coletar_, 0);
+       pp(_eaction1.dados_, 0);
+       if (_i_ > 0) render(_R_PAREN);
+    }
+    else     if (foo instanceof Centralizador.Absyn.EAction2)
+    {
+       Centralizador.Absyn.EAction2 _eaction2 = (Centralizador.Absyn.EAction2) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       pp(_eaction2.aplicar_, 0);
+       pp(_eaction2.medicacao_, 0);
+       if (_i_ > 0) render(_R_PAREN);
+    }
+    else     if (foo instanceof Centralizador.Absyn.EAction3)
+    {
+       Centralizador.Absyn.EAction3 _eaction3 = (Centralizador.Absyn.EAction3) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       render("Autodestruicao");
+       if (_i_ > 0) render(_R_PAREN);
+    }
+  }
+
+  private static void pp(Centralizador.Absyn.Coletar foo, int _i_)
+  {
+    if (foo instanceof Centralizador.Absyn.ECollect1)
+    {
+       Centralizador.Absyn.ECollect1 _ecollect1 = (Centralizador.Absyn.ECollect1) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       render("Iniciar Medicao de ");
+       if (_i_ > 0) render(_R_PAREN);
+    }
+    else     if (foo instanceof Centralizador.Absyn.ECollect2)
+    {
+       Centralizador.Absyn.ECollect2 _ecollect2 = (Centralizador.Absyn.ECollect2) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       render("Parar Medicao de ");
+       if (_i_ > 0) render(_R_PAREN);
+    }
+  }
+
+  private static void pp(Centralizador.Absyn.Aplicar foo, int _i_)
+  {
+    if (foo instanceof Centralizador.Absyn.EApply1)
+    {
+       Centralizador.Absyn.EApply1 _eapply1 = (Centralizador.Absyn.EApply1) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       render("Liberar");
+       if (_i_ > 0) render(_R_PAREN);
+    }
+    else     if (foo instanceof Centralizador.Absyn.EApply2)
+    {
+       Centralizador.Absyn.EApply2 _eapply2 = (Centralizador.Absyn.EApply2) foo;
+       if (_i_ > 0) render(_L_PAREN);
+       render("Cessar Liberacao de ");
+       if (_i_ > 0) render(_R_PAREN);
+    }
+  }
+
+  private static void pp(Centralizador.Absyn.Dados foo, int _i_)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.